### ISSUE TITLE
Unable to craco build in Heroku or netlify 

### ISSUE THREAD

**Issue by DavidSalemLee** [gsoft-inc/craco#328]
I'm using craco with tailwindcss and trying to deploy to heroku and/or netlify and both are failing build. It gives me no problems locally but on deployment, it just will not build. 

Netlify error:
```
2:33:14 PM: Build ready to start
2:33:16 PM: build-image version: 081db65c3e4ce8423fedb40e7689a87de6f84667
2:33:16 PM: build-image tag: v4.3.1
2:33:16 PM: buildbot version: 1523fa12d6de79e532ed0aff71973c5fbb8e58aa
2:33:16 PM: Fetching cached dependencies
2:33:16 PM: Failed to fetch cache, continuing with build
2:33:16 PM: Starting to prepare the repo for build
2:33:16 PM: No cached dependencies found. Cloning fresh repo
2:33:18 PM: Preparing Git Reference refs/heads/main
2:33:19 PM: Parsing package.json dependencies
2:33:19 PM: Different publish path detected, going to use the one specified in the Netlify configuration file: 'frontend' versus '/' in the Netlify UI
2:33:19 PM: Starting build script
2:33:19 PM: Installing dependencies
2:33:19 PM: Python version set to 2.7
2:33:20 PM: Downloading and installing node v16.10.0...
2:33:20 PM: Downloading https://nodejs.org/dist/v16.10.0/node-v16.10.0-linux-x64.tar.xz...
2:33:21 PM: Computing checksum with sha256sum
2:33:21 PM: Checksums matched!
2:33:24 PM: Now using node v16.10.0 (npm v7.24.0)
2:33:24 PM: Started restoring cached build plugins
2:33:24 PM: Finished restoring cached build plugins
2:33:24 PM: Attempting ruby version 2.7.2, read from environment
2:33:25 PM: Using ruby version 2.7.2
2:33:25 PM: Using PHP version 8.0
2:33:25 PM: Started restoring cached node modules
2:33:25 PM: Finished restoring cached node modules
2:33:26 PM: Installing NPM modules using NPM version 7.24.0
2:33:28 PM: npm WARN old lockfile
2:33:28 PM: npm WARN old lockfile The package-lock.json file was created with an old version of npm,
2:33:28 PM: npm WARN old lockfile so supplemental metadata must be fetched from the registry.
2:33:28 PM: npm WARN old lockfile
2:33:28 PM: npm WARN old lockfile This is a one-time fix-up, please be patient...
2:33:28 PM: npm WARN old lockfile
2:33:41 PM: npm WARN old lockfile Error: Integrity checksum failed when using sha512: wanted sha512-k+f6DL1KxG0d2dEUILxytWpHRBPMagQ8wUdH5bJUzbmv7TPd3yriR+nSOXy2Hcvt79cP3Lgc5J6oPjO4PNyRLg== but got sha512-EireCtpQyyJ4Xz8NYzHafBoy4baCOO96flM0+HgtsFcIQ9KFy/YBK3GEtlnD+rXen0e4xm8t3WiUcKBJmN6yjg==.
2:33:41 PM: npm WARN old lockfile     at /opt/buildhome/.nvm/versions/node/v16.10.0/lib/node_modules/npm/node_modules/pacote/lib/registry.js:145:39
2:33:41 PM: npm WARN old lockfile     at async Array.<anonymous> (/opt/buildhome/.nvm/versions/node/v16.10.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:717:9)
2:33:41 PM: npm WARN old lockfile  Could not fetch metadata for tailwindcss@2.2.16 Error: Integrity checksum failed when using sha512: wanted sha512-k+f6DL1KxG0d2dEUILxytWpHRBPMagQ8wUdH5bJUzbmv7TPd3yriR+nSOXy2Hcvt79cP3Lgc5J6oPjO4PNyRLg== but got sha512-EireCtpQyyJ4Xz8NYzHafBoy4baCOO96flM0+HgtsFcIQ9KFy/YBK3GEtlnD+rXen0e4xm8t3WiUcKBJmN6yjg==.
2:33:41 PM: npm WARN old lockfile     at /opt/buildhome/.nvm/versions/node/v16.10.0/lib/node_modules/npm/node_modules/pacote/lib/registry.js:145:39
2:33:41 PM: npm WARN old lockfile     at async Array.<anonymous> (/opt/buildhome/.nvm/versions/node/v16.10.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:717:9) {
2:33:41 PM: npm WARN old lockfile   code: 'EINTEGRITY'
2:33:41 PM: npm WARN old lockfile }
2:33:54 PM: npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
2:33:55 PM: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
2:33:55 PM: npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
2:33:56 PM: npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2:33:56 PM: npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2:33:58 PM: npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
2:33:58 PM: npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
2:33:59 PM: npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
2:34:05 PM: npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
2:34:07 PM: npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
2:34:07 PM: npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
2:34:08 PM: npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
2:34:09 PM: npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
2:34:09 PM: npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
2:34:10 PM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
2:34:10 PM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
2:34:12 PM: npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
2:34:17 PM: npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
2:34:19 PM: added 2038 packages, and audited 2039 packages in 52s
2:34:19 PM: 154 packages are looking for funding
2:34:19 PM:   run `npm fund` for details
2:34:19 PM: 11 moderate severity vulnerabilities
2:34:19 PM: To address all issues (including breaking changes), run:
2:34:19 PM:   npm audit fix --force
2:34:19 PM: Run `npm audit` for details.
2:34:19 PM: NPM modules installed
2:34:19 PM: Started restoring cached go cache
2:34:19 PM: Finished restoring cached go cache
2:34:19 PM: go version go1.16.5 linux/amd64
2:34:19 PM: go version go1.16.5 linux/amd64
2:34:19 PM: Installing missing commands
2:34:19 PM: Verify run directory
2:34:21 PM: ​
2:34:21 PM: ────────────────────────────────────────────────────────────────
2:34:21 PM:   Netlify Build                                                 
2:34:21 PM: ────────────────────────────────────────────────────────────────
2:34:21 PM: ​
2:34:21 PM: ❯ Version
2:34:21 PM:   @netlify/build 18.13.6
2:34:21 PM: ​
2:34:21 PM: ❯ Flags
2:34:21 PM:   baseRelDir: true
2:34:21 PM:   buildId: 6157627a67eea9281106405a
2:34:21 PM:   deployId: 6157627a67eea9281106405c
2:34:21 PM: ​
2:34:21 PM: ❯ Current directory
2:34:21 PM:   /opt/build/repo/frontend
2:34:21 PM: ​
2:34:21 PM: ❯ Config file
2:34:21 PM:   No config file was defined: using default values.
2:34:21 PM: ​
2:34:21 PM: ❯ Context
2:34:21 PM:   production
2:34:21 PM: ​
2:34:21 PM: ────────────────────────────────────────────────────────────────
2:34:21 PM:   1. Build command from Netlify app                             
2:34:21 PM: ────────────────────────────────────────────────────────────────
2:34:21 PM: ​
2:34:21 PM: $ npm run build
2:34:21 PM: > frontend@0.1.0 build
2:34:21 PM: > craco build --verbose
2:34:21 PM: craco:  Project root path resolved to:  /opt/build/repo/frontend
2:34:21 PM: craco:  Config file path resolved to:  /opt/build/repo/frontend/craco.config.js
2:34:21 PM: craco:  Override started with arguments:  [
2:34:21 PM:   '/opt/buildhome/.nvm/versions/node/v16.10.0/bin/node',
2:34:21 PM:   '/opt/build/repo/frontend/node_modules/@craco/craco/scripts/build.js',
2:34:21 PM:   '--verbose'
2:34:21 PM: ]
2:34:21 PM: craco:  For environment:  production
2:34:21 PM: craco:  Found craco config file at:  /opt/build/repo/frontend/craco.config.js
2:34:22 PM: craco:  Applied craco config plugins.
2:34:22 PM: craco:  Found Webpack prod config at:  /opt/build/repo/frontend/node_modules/react-scripts/config/webpack.config.js
2:34:23 PM: craco:  Overrided ESLint config to enable an ignore file.
2:34:24 PM: craco:  Added PostCSS plugins.
2:34:24 PM: craco:  Overrided PostCSS loader.
2:34:24 PM: craco:  Added PostCSS plugins.
2:34:24 PM: craco:  Overrided PostCSS loader.
2:34:24 PM: craco:  Added PostCSS plugins.
2:34:24 PM: craco:  Overrided PostCSS loader.
2:34:24 PM: craco:  Added PostCSS plugins.
2:34:24 PM: craco:  Overrided PostCSS loader.
2:34:24 PM: craco:  Applied webpack config plugins.
2:34:24 PM: craco:  Overrided require cache for module: /opt/build/repo/frontend/node_modules/react-scripts/config/webpack.config.js
2:34:24 PM: craco:  Overrided Webpack prod config.
2:34:24 PM: craco:  Building CRA at:  /opt/build/repo/frontend/node_modules/react-scripts/scripts/build.js
2:34:24 PM: Creating an optimized production build...
2:34:28 PM: Failed to compile.
2:34:28 PM: 
2:34:28 PM: ./src/App.js
2:34:28 PM: Cannot find file './screens/CartScreen' in './src'.
2:34:28 PM: ​
2:34:28 PM: ────────────────────────────────────────────────────────────────
2:34:28 PM:   "build.command" failed                                        
2:34:28 PM: ────────────────────────────────────────────────────────────────
2:34:28 PM: ​
2:34:28 PM:   Error message
2:34:28 PM:   Command failed with exit code 1: npm run build
2:34:28 PM: ​
2:34:28 PM:   Error location
2:34:28 PM:   In Build command from Netlify app:
2:34:28 PM:   npm run build
2:34:28 PM: ​
2:34:28 PM:   Resolved config
2:34:28 PM:   build:
2:34:28 PM:     base: /opt/build/repo/frontend
2:34:28 PM:     command: npm run build
2:34:28 PM:     commandOrigin: ui
2:34:28 PM:     publish: /opt/build/repo/frontend
2:34:28 PM:     publishOrigin: ui
2:34:28 PM: Caching artifacts
2:34:28 PM: Started saving node modules
2:34:28 PM: Finished saving node modules
2:34:28 PM: Started saving build plugins
2:34:28 PM: Finished saving build plugins
2:34:28 PM: Started saving pip cache
2:34:28 PM: Finished saving pip cache
2:34:28 PM: Started saving emacs cask dependencies
2:34:28 PM: Finished saving emacs cask dependencies
2:34:28 PM: Started saving maven dependencies
2:34:28 PM: Finished saving maven dependencies
2:34:28 PM: Started saving boot dependencies
2:34:29 PM: Finished saving boot dependencies
2:34:29 PM: Started saving rust rustup cache
2:34:29 PM: Finished saving rust rustup cache
2:34:29 PM: Started saving go dependencies
2:34:29 PM: Finished saving go dependencies
2:34:31 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
2:34:31 PM: Creating deploy upload records
2:34:31 PM: Failing build: Failed to build site
2:34:31 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2
2:34:31 PM: Finished processing build request in 1m15.059036695s
```

Heroku build log:
```


-----> Building on the Heroku-20 stack

-----> Using buildpack: mars/create-react-app

-----> React.js (create-react-app) multi app detected

=====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-multi.git

=====> Detected Framework: Multipack

=====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-nodejs.git

=====> Detected Framework: Node.js

       

-----> Creating runtime environment

       

       NPM_CONFIG_LOGLEVEL=error

       NODE_VERBOSE=false

       NODE_ENV=production

       NODE_MODULES_CACHE=true

       

-----> Installing binaries

       engines.node (package.json):  14.17.5

       engines.npm (package.json):   unspecified (use default)

       

       Resolving node version 14.17.5...

       Downloading and installing node 14.17.5...

       Using default npm version: 6.14.14

       

-----> Restoring cache

       - node_modules

       

-----> Installing dependencies

       Installing node modules

       

       > nodemon@2.0.12 postinstall /tmp/build_c9292af6/node_modules/nodemon

       > node bin/postinstall || exit 0

       

       Love nodemon? You can now support the project via the open collective:

        > https://opencollective.com/nodemon/donate

       

       added 366 packages in 14.439s

       

-----> Build

       Running heroku-postbuild

       

       > react-ecommerce@1.0.0 heroku-postbuild /tmp/build_c9292af6

       > NPM_CONFIG_PRODUCTION=false npm install --prefix frontend && npm run build --prefix frontend

       

       

       > core-js@2.6.12 postinstall /tmp/build_c9292af6/frontend/node_modules/babel-runtime/node_modules/core-js

       > node -e "try{require('./postinstall')}catch(e){}"

       

       

       > core-js@3.16.3 postinstall /tmp/build_c9292af6/frontend/node_modules/core-js

       > node -e "try{require('./postinstall')}catch(e){}"

       

       

       > core-js-pure@3.16.3 postinstall /tmp/build_c9292af6/frontend/node_modules/core-js-pure

       > node -e "try{require('./postinstall')}catch(e){}"

       

       

       > ejs@2.7.4 postinstall /tmp/build_c9292af6/frontend/node_modules/ejs

       > node ./postinstall.js

       

       added 1991 packages from 811 contributors in 54.26s

       

       150 packages are looking for funding

         run `npm fund` for details

       

       

       > frontend@0.1.0 build /tmp/build_c9292af6/frontend

       > craco build

       

       Creating an optimized production build...

       Failed to compile.

       

       ./src/App.js

       Cannot find file './screens/CartScreen' in './src'.

       

       

npm ERR! code ELIFECYCLE

npm ERR! errno 1

npm ERR! frontend@0.1.0 build: `craco build`

npm ERR! Exit status 1

npm ERR! 

npm ERR! Failed at the frontend@0.1.0 build script.

npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

npm ERR!     /tmp/npmcache.t46Vn/_logs/2021-10-01T18_15_55_702Z-debug.log

npm ERR! code ELIFECYCLE

npm ERR! errno 1

npm ERR! react-ecommerce@1.0.0 heroku-postbuild: `NPM_CONFIG_PRODUCTION=false npm install --prefix frontend && npm run build --prefix frontend`

npm ERR! Exit status 1

npm ERR! 

npm ERR! Failed at the react-ecommerce@1.0.0 heroku-postbuild script.

npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

npm ERR!     /tmp/npmcache.t46Vn/_logs/2021-10-01T18_15_55_728Z-debug.log

-----> Build failed

       

       We're sorry this build is failing! You can troubleshoot common issues here:

       https://devcenter.heroku.com/articles/troubleshooting-node-deploys

       

       If you're stuck, please submit a ticket so we can help:

       https://help.heroku.com/

       

       Love,

       Heroku

       

 !     Push rejected, failed to compile React.js (create-react-app) multi app.

 !     Push failed

```

Package.json 

```
{
  "name": "frontend",
  "proxy": "http://127.0.0.1:5000",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@craco/craco": "^6.2.0",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^11.2.7",
    "@testing-library/user-event": "^12.8.3",
    "axios": "^0.21.1",
    "react": "^17.0.2",
    "react-bootstrap": "^1.6.1",
    "react-dom": "^17.0.2",
    "react-paypal-button-v2": "^2.6.3",
    "react-redux": "^7.2.5",
    "react-router-dom": "^5.2.1",
    "react-scripts": "4.0.3",
    "redux": "^4.1.1",
    "redux-devtools-extension": "^2.13.9",
    "redux-thunk": "^2.3.0",
    "web-vitals": "^1.1.2"
  },
  "scripts": {
    "start": "craco start --verbose",
    "build": "craco build --verbose",
    "test": "craco test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "autoprefixer": "^9.8.7",
    "postcss": "^7.0.38",
    "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.2.16"
  }
}
```
I've tried moving tailwindcss to dependencies and still does not work. 

Stumped! Can anyone help?

**Comment by DavidSalemLee** [comment]
I need to add that while it says "Cannot find file './screens/CartScreen' in './src' " that's not the problem it seems as that file is there and it works perfectly fine locally. The app is finished. Deployment is doing something funky and I'm not sure what the problem is here that both Heroku and Netlify have issues with craco build. 

If I move tailwindcss install around sometimes it will give me different error called "module not found:tailwindcss". I found another thread that had that problem but for me it didn't fix it. 

I've tried uninstalling reinstalling tailwind packages, been dealing with this for 3 days now..

**Comment by stalha97** [comment]
Hi, I am facing the same issues. Were you able to resolve it?

**Comment by stalha97** [comment]
@DavidSalemLee Hi, were you able to solve the issue? I would like to know the fix as well.

**Comment by DavidSalemLee** [comment]
@stalha97 I had two concurrent problems which was why it was tricky to fix. 1, tailwind package has to be moved out of dev dependency into regular dependency along with postcss, etc. 

Then I got cartScreen.js not found which was a glitch in my github file. For some reason cartScreen was not changed to CartScreen.js through my changes on github and I had to manually edit the filename in github and once I did that heroku was able to build successfully
